packages/cli/src/commands/deploy/action.ts

```diff
@@ -0,0 +1,63 @@
+import { TxParams } from '@openzeppelin/upgrades';
+import { compile } from '../../models/compiler/Compiler';
+import { fromContractFullName } from '../../utils/naming';
+import ConfigManager from '../../models/config/ConfigManager';
+import NetworkController from '../../models/network/NetworkController';
+import stdout from '../../utils/stdout';
+import { MethodArgType } from '../../prompts/prompt';
+import { parseMultipleArgs } from '../../utils/input';
+import NetworkFile from '../../models/files/NetworkFile';
+import { createAction } from '../create';
+
+import { AbortAction } from '../../register-command';
+
+interface Options {
+  network: string;
+  skipCompile?: boolean;
+  upgradeable?: boolean;
+
+  // The following are not available as CLI flags, and they are only used in tests.
+  networkFile?: NetworkFile;
+  txParams?: TxParams;
+}
+
+export async function preAction(
+  contractName: string | undefined,
+  deployArgs: string[],
+  options: Options,
+): Promise<void> {
+  if (!options.skipCompile) {
+    await compile();
+  }
+
+  if (options.upgradeable === true) {
+    throw new AbortAction(async () => {
```

Comment:
Yes, good catch. I don't see a way to make that happen without building it in in a very ad-hoc way. However, aborting early is only important now because we have to delegate to `create`. Once we add the transpiler it will be built in to `deploy`, and it won't be necessary to do this abort thing. In fact, I would remove the abort feature once we don't need it anymore.